From 04932935a3f204b02290826da36ff6ccf1f8d920 Mon Sep 17 00:00:00 2001
From: chalesyu <574249312@qq.com>
Date: Sun, 14 Mar 2021 00:00:00 +0800
Subject: [PATCH] Add support for A63

---
 soc_info.c                | 14 ++++++++++++++
 uart0-helloworld-sdboot.c | 13 ++++++++++---
 2 files changed, 24 insertions(+), 3 deletions(-)

diff --git a/soc_info.c b/soc_info.c
index 6414fa7..d695a7c 100644
--- a/soc_info.c
+++ b/soc_info.c
@@ -307,6 +307,20 @@ soc_info_t soc_info_table[] = {
 		/* Check L.NOP in the OpenRISC reset vector */
 		.needs_smc_workaround_if_zero_word_at_addr = 0x100004,
 		.watchdog     = &wd_h6_compat,
+	},{
+		.soc_id       = 0x1719, /* Allwinner A63 */
+		.name         = "A63",
+		.spl_addr     = 0x20000,
+		.scratch_addr = 0x21000,
+		.thunk_addr   = 0x42200, .thunk_size = 0x200,
+		.swap_buffers = h6_sram_swap_buffers,
+		.sram_size    = 144 * 1024,
+		.sid_base     = 0x03006000,
+		.sid_offset   = 0x200,
+		.rvbar_reg    = 0x09010040,
+		/* Check L.NOP in the OpenRISC reset vector */
+		.needs_smc_workaround_if_zero_word_at_addr = 0x100004,
+		.watchdog     = &wd_h6_compat,
 	},{
 		.soc_id       = 0x1817, /* Allwinner V831 */
 		.name         = "V831",
diff --git a/uart0-helloworld-sdboot.c b/uart0-helloworld-sdboot.c
index 52d7e43..4465b93 100644
--- a/uart0-helloworld-sdboot.c
+++ b/uart0-helloworld-sdboot.c
@@ -306,6 +306,7 @@ void soc_detection_init(void)
 #define soc_is_a80()	(soc_id == 0x1639)
 #define soc_is_a64()	(soc_id == 0x1689)
 #define soc_is_h5()	(soc_id == 0x1718)
+#define soc_is_a63()	(soc_id == 0x1719)
 #define soc_is_h6()	(soc_id == 0x1728)
 #define soc_is_h616()	(soc_id == 0x1823)
 #define soc_is_r40()	(soc_id == 0x1701)
@@ -386,7 +387,7 @@ void clock_init_uart_h6(void)
 
 void clock_init_uart(void)
 {
-	if (soc_is_h6() || soc_is_v831() || soc_is_h616())
+	if (soc_is_h6() || soc_is_a63() || soc_is_v831() || soc_is_h616())
 		clock_init_uart_h6();
 	else
 		clock_init_uart_legacy();
@@ -432,6 +433,10 @@ void gpio_init(void)
 		sunxi_gpio_set_cfgpin(SUNXI_GPA(4), SUN50I_H5_GPA_UART0);
 		sunxi_gpio_set_cfgpin(SUNXI_GPA(5), SUN50I_H5_GPA_UART0);
 		sunxi_gpio_set_pull(SUNXI_GPA(5), SUNXI_GPIO_PULL_UP);
+	} else if (soc_is_a63()) {
+		sunxi_gpio_set_cfgpin(SUNXI_GPB(9), SUN50I_A64_GPB_UART0);
+		sunxi_gpio_set_cfgpin(SUNXI_GPB(10), SUN50I_A64_GPB_UART0);
+		sunxi_gpio_set_pull(SUNXI_GPB(10), SUNXI_GPIO_PULL_UP);
 	} else if (soc_is_h6()) {
 		sunxi_gpio_set_cfgpin(SUNXI_GPH(0), SUN50I_H6_GPH_UART0);
 		sunxi_gpio_set_cfgpin(SUNXI_GPH(1), SUN50I_H6_GPH_UART0);
@@ -524,7 +529,7 @@ int get_boot_device(void)
 	u32 *spl_signature = (void *)0x4;
 	if (soc_is_a64() || soc_is_a80() || soc_is_h5())
 		spl_signature = (void *)0x10004;
-	if (soc_is_h6() || soc_is_v831() || soc_is_h616())
+	if (soc_is_h6() || soc_is_a63() || soc_is_v831() || soc_is_h616())
 		spl_signature = (void *)0x20004;
 
 	/* Check the eGON.BT0 magic in the SPL header */
@@ -542,7 +547,7 @@ int get_boot_device(void)
 
 void bases_init(void)
 {
-	if (soc_is_h6() || soc_is_v831() || soc_is_h616()) {
+	if (soc_is_h6() || soc_is_a63() || soc_is_v831() || soc_is_h616()) {
 		pio_base = H6_PIO_BASE;
 		uart0_base = H6_UART0_BASE;
 	} else {
@@ -577,6 +582,8 @@ int main(void)
 		uart0_puts("Allwinner H3!\n");
 	else if (soc_is_h5())
 		uart0_puts("Allwinner H5!\n");
+	else if (soc_is_a63())
+		uart0_puts("Allwinner A63!\n");
 	else if (soc_is_h6())
 		uart0_puts("Allwinner H6!\n");
 	else if (soc_is_h616())
-- 

