diff --git a/drivers/net/usb/sr9700.c b/drivers/net/usb/sr9700.c
index b658510..0907d4e 100644
--- a/drivers/net/usb/sr9700.c
+++ b/drivers/net/usb/sr9700.c
@@ -353,10 +353,15 @@ static int sr9700_bind(struct usbnet *dev, struct usb_interface *intf)
 	 */
 	if (sr_read(dev, SR_PAR, ETH_ALEN, addr) < 0) {
 		netdev_err(netdev, "Error reading MAC address\n");
-		ret = -ENODEV;
-		goto out;
+
+		/* Give it a random MAC address when reading failed */
+		eth_random_addr(netdev->dev_addr);
+		netdev_err(netdev,
+			 "Invalid MAC address, using random address %pM\n",
+			 netdev->dev_addr);
+		sr_write_async(dev, SR_PAR, 6, netdev->dev_addr);
 	}
-	eth_hw_addr_set(netdev, addr);
+	else eth_hw_addr_set(netdev, addr);
 
 	/* power up and reset phy */
 	sr_write_reg(dev, SR_PRR, PRR_PHY_RST);
@@ -536,6 +541,10 @@ static const struct usb_device_id products[] = {
 		USB_DEVICE(0x0fe6, 0x9700),	/* SR9700 device */
 		.driver_info = (unsigned long)&sr9700_driver_info,
 	},
+	{
+		USB_DEVICE(0x0fe6, 0x9702),	/* KY-RD9700 with CD-ROM */
+		.driver_info = (unsigned long)&sr9700_driver_info,
+	 },
 	{},			/* END */
 };
 
--

